org.mixedrealitytoolkit.spatialmanipulation/ObjectManipulator/MoveLogics/ScaleLogic.cs

```diff
@@ -78,8 +78,12 @@ private float GetScaleBetweenInteractors(List<IXRSelectInteractor> interactors,
                 for (int j = i + 1; j < interactors.Count; j++)
                 {
                     // Defer square root until end for performance.
-                    var distance = Vector3.SqrMagnitude(interactors[i].transform.position -
-                                                       interactors[j].transform.position);
+                    var distance = Vector3.SqrMagnitude(interactors[i].GetAttachTransform(interactable).position -
```

Comment:
How do we get into the case where two interactors have exactly identical positions? And is there any hint as to why it's seemingly only seen on Quest?

Reply:
@keveleigh   This only seems to occur with XRI 3 and when using the updated default XRI rig in the scene instead of our MRTK3 rig.  The issue describes reproduction using XRI's HandsDemoScene sample, and I was able to reproduce with the XRI rig instead of ours.

In the XRI 3 default rig, for the hand interactor groups, they no longer modify the position of the interactor (Near-Far Interactor) itself using something like a tracked pose driver.  Instead, that interactor utilizes a InteractionAttachController component, which creates new GameObjects to represent the attach points (anchor and child) for the interaction, themselves getting positional updates.

So, in this case, the interactors themselves not getting transform updates, had matching positions.  I know this isn't an issue with our rig, but thought it was worth protecting against cases like this where the interactor positions could match.

Reply:
@keveleigh @AMollis 

Also, the fact that XRI 3 is updating their systems to use things like the InteractionAttachController, which creates sub objects to represent transforms for the interaction, instead of updating the interactor transform itself, may possibly cause issues when used in combination with MRTK3's logic in other areas (like this case with ScaleLogic).

I couldn't think of other places that may be impacted off hand, but thought there might be since a moving interactor is an assumption we make generally I think.  Worth discussing?